src/sensai/util/cache.py

```diff
@@ -576,8 +578,45 @@ def __init__(self, cacheBasePath: str, filenamePrefix: str = None, filename: str
         else:
             self.filenamePrefix += "-"
 
-    def __call__(self, fn, *args, **kwargs):
+    def __call__(self, fn: Callable, *args, **kwargs):
         if self.filename is None:
             self.filename = self.filenamePrefix + fn.__qualname__ + ".cache.pickle"
         picklePath = os.path.join(self.cacheBasePath,  self.filename)
         return lambda *args, **kwargs: cached(lambda: fn(*args, **kwargs), picklePath, functionName=fn.__name__, backend=self.backend)
+
+
+class LoadSaveInterface(ABC):
+    @abstractmethod
+    def save(self, path: str) -> None:
+        pass
+
+    @classmethod
+    @abstractmethod
+    def load(cls: T, path: str) -> T:
+        pass
+
+
+class PickleSerializingMixin(LoadSaveInterface):
```

Comment:
Well, it's using pickle in particular. I have another class that implements the same interface with geopandas (saving to and loading from a shapefile)

Reply:
OK, I assumed there would (probably) only ever be pickle. In that case, how about PickleLoadSaveMixin?
I don't like "Serializing" because (a) it uses American spelling and (b) it is not as specific with regard to which functions are being mixed in (and strictly speaking does not even mention deserialisation). But that's just me.

Reply:
Done